### DISCUSSION TITLE
Improving the error message when drawing methods are used outside setup/draw (and not in static mode)?

### DISCUSSION THREAD

**Discussion by villares** [py5coding/py5generator#487] [Ideas]
As @vsquared has shown initially at https://github.com/py5coding/py5generator/discussions/486, the error message `Cannot invoke "processing.core.PGraphics.line(float, float, float, float)" because "this.g" is null` can be a bit confusing for those getting started with py5/Processing. Maybe we could add to it some complementary friendly advice?

**Comment by hx2A** [comment]
py5's error handling is activate for running py5 Sketches. If a user attempts to call a py5 method outside of a running Sketch, the exception is not caught by anything py5 can use to intervene and provide a better message. I agree that the message is confusing. I will need to think about what can be done here.

**Reply by villares** [reply]
I see the problem :(

**Reply by hx2A** [reply]
I added it to my list of things to investigate. When I come up with ideas or a solution I'll let you know!
